diff --git a/csh.login b/csh.login
index 51902e8..e4b1990 100644
--- a/csh.login
+++ b/csh.login
@@ -2,20 +2,28 @@
 
 # System wide environment and startup programs, for login setup
 
-#add sbin directories to the path
-foreach p ( /usr/local/sbin /usr/sbin )
-	switch (":${PATH}:")
-	case "*:${p}:*":
-		breaksw
-	default:
-		if ( $uid == 0 ) then
-                        set path = ( ${p} ${path:q} )
-		else
-                        set path = ( ${path:q} ${p} )
-		endif
-		breaksw
-	endsw
-end
+if ( ! ${?PATH} ) then
+        if ( $uid == 0 ) then
+		setenv PATH "/usr/local/sbin:/usr/sbin:/usr/local/bin:/usr/bin"
+        else
+		setenv PATH "/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin"
+        endif
+else
+	#add sbin directories to the path
+	foreach p ( /usr/local/sbin /usr/sbin )
+		switch (":${PATH}:")
+		case "*:${p}:*":
+			breaksw
+		default:
+			if ( $uid == 0 ) then
+	                        set path = ( ${p} ${path:q} )
+			else
+	                        set path = ( ${path:q} ${p} )
+			endif
+			breaksw
+		endsw
+	end
+endif
 
 setenv HOSTNAME `/usr/bin/hostname`
 set history=1000
