diff --git a/services b/services
index b9eb06c..470bf86 100644
--- a/services
+++ b/services
@@ -8828,7 +8828,7 @@ cssc            5637/tcp                # Symantec CSSC
 flcrs           5638/tcp                # Symantec Fingerprint Lookup and Container Reference
 ics             5639/tcp                # Symantec Integrity Checking
 vfmobile        5646/tcp                # Ventureforth Mobile
-###UNAUTHORIZED USE: Port 5666 used by SAIC NRPE############
+nrpe            5666/tcp                # Nagios Remote Plugin Executor
 filemq          5670/tcp                # ZeroMQ file
 zre-disc        5670/udp                # Local area discovery and msging over ZeroMQ
 amqps           5671/tcp                # amqp protocol over TLS/SSL
